tools/hilbert.py

```diff
@@ -289,23 +289,91 @@ def cmd_list_applications(parser, context, args):
     group.add_argument('--configdump', required=False,
                        help="specify input dump file")
 
+    parser.add_argument('-f', '--format', choices=['list', 'dashboard'],
+                        default='list', required=False,
+                        help="output format")
+
     args = parser.parse_args(args)
 
-    log.debug("Listing all Application ID...")
+    _args = vars(args)
+
+    mode = _args.get('format', 'list')
+
+    if mode == "list":
+        log.debug("Listing all Application ID...")
+
+        obj = None
+        try:
+            obj = cmd_list(parser, context, args, 'Applications/keys')
+        except:
+            log.exception("Sorry could not get the list of '{}' from the input file!".format('applications'))
+            sys.exit(1)
+
+        assert obj is not None
+
+        if isinstance(obj, BaseValidator):
+            print(yaml_dump(obj.data_dump()))
+        else:
+            print(yaml_dump(obj))
+
+        return args
+
+    assert mode == 'dashboard'
+
+    log.debug("Loading/parsing configuration...")
+    cfg = input_handler(parser, vars(context), args)
+    assert cfg is not None
+    assert isinstance(cfg, Hilbert)
 
     obj = None
+    log.debug("Listing all Applications for Dashboard...")
     try:
-        obj = cmd_list(parser, context, args, 'Applications/keys')
+        obj = cfg.query('Applications/data')
     except:
-        log.exception("Sorry could not get the list of '{}' from the input file!".format('applications'))
+        log.exception("Sorry could not query Applications's details from the given configuration!")
         sys.exit(1)
 
     assert obj is not None
-
     if isinstance(obj, BaseValidator):
-        print(yaml_dump(obj.data_dump()))
-    else:
-        print(yaml_dump(obj))
+        obj = obj.data_dump()
+
+    first = True
+    print('[')
+
+
+    for k in obj:
+        o = obj.get(k, None)
+        assert o is not None
+
+        sts = ""
+        if 'compatible_stations' in o:
+            try:
+                sts = o.get('compatible_stations', None)
+                if isinstance(sts, BaseValidator):
+                    sts = sts.data_dump()
+                sts = '"' + '", "'.join(sts) + '"'
+                if sts.strip() == '""':
+                    sts = ""
+            except:
+                log.debug("No [compatible_stations] for application [{0}]".format(k))
+
+        if not first:
+            print(',')
+
+        print('{')
+        for f in ['name', 'description', 'icon']:
```

Comment:
Why not just output all the available information? (Again, this is too specific too the dashboard.)

Reply:
it was all done for the Dashboard. Some fields make absolutely no sense for it.